app/Http/Controllers/User/EmailValidationController.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Http\Controllers\User;
+
+use App\Http\Controllers\Controller;
+use App\Models\UserValidation;
+use App\User;
+use Illuminate\Http\Request;
+
+class EmailValidationController extends Controller
+{
+    public function validateEmail(Request $request, User $user)
```

Comment:
Email providers **will** open all links in emails to check for spam/fishing, so a user receiving the code will have very weird stuff happen.

The page should only work if the user is logged in to avoid this.